pkg/cmd/roachtest/operations/node_kill.go

```diff
@@ -20,19 +20,10 @@ import (
 )
 
 type cleanupNodeKill struct {
-	nodes    option.NodeListOption
-	downtime time.Duration
+	nodes option.NodeListOption
 }
 
 func (cl *cleanupNodeKill) Cleanup(ctx context.Context, o operation.Operation, c cluster.Cluster) {
-	o.Status(fmt.Sprintf("waiting for %s before restarting node %s", cl.downtime, cl.nodes))
-	select {
-	case <-time.After(cl.downtime):
-	case <-ctx.Done():
-		o.Status("cleanup context cancelled during wait")
-		return
-	}
-
 	db, err := c.ConnE(ctx, o.L(), cl.nodes[0])
 	if err != nil {
 		err = c.RunE(ctx, option.WithNodes(cl.nodes), "./cockroach.sh")
```

Comment:
Drive-by comment, I'm a bit surprised we've been using adhoc commands to start/stop nodes instead of `roachprod.Start` and `roachprod.Stop`.

Reply:
It's due to the problem described here: https://github.com/cockroachdb/cockroach/pull/123517

I looked into it though, the fix should be as simple as having a fall back case if service registration is not enabled. For example, see my WIP change in https://github.com/cockroachdb/cockroach/blob/e1549cf92545178cd580ed86603e74e42cff0d2e/pkg/roachprod/install/cluster_synced.go#L461-L488.